R/LearnerAvg.R

```diff
@@ -163,35 +191,45 @@ LearnerRegrAvg = R6Class("LearnerRegrAvg", inherit = LearnerRegr,
 mlr_learners_regr.avg = NULL
 mlr_learners_classif.avg = NULL
 
-nlopt_levels = c("NLOPT_GN_DIRECT", "NLOPT_GN_DIRECT_L", "NLOPT_GN_DIRECT_L_RAND",
-  "NLOPT_GN_DIRECT_NOSCAL", "NLOPT_GN_DIRECT_L_NOSCAL", "NLOPT_GN_DIRECT_L_RAND_NOSCAL",
-  "NLOPT_GN_ORIG_DIRECT", "NLOPT_GN_ORIG_DIRECT_L", "NLOPT_LN_PRAXIS",
-  "NLOPT_GN_CRS2_LM", "NLOPT_GN_MLSL", "NLOPT_GN_MLSL_LDS",
-  "NLOPT_LN_COBYLA", "NLOPT_LN_NEWUOA", "NLOPT_LN_NEWUOA_BOUND",
-  "NLOPT_LN_NELDERMEAD", "NLOPT_LN_SBPLX", "NLOPT_LN_AUGLAG",
-  "NLOPT_LN_AUGLAG_EQ", "NLOPT_LN_BOBYQA", "NLOPT_GN_ISRES")
-
-nlopt_objfun = function(weights, task, measure, avg_weight_fun, data) {
-  # This is the objective function we minimize using nlopt
-  prd = avg_weight_fun(task, weights, data)
-  res = prd$score(measure)
-  if (measure$minimize) res else -res
-}
+optimize_weights_learneravg = function(self, task, n_weights, data) {
 
-optimize_objfun_nlopt = function(task, pars, avg_weight_fun, n_weights, data) {
-  require_namespaces("nloptr")
-  measure = assert_measure(as_measure(pars$measure, task_type = task$task_type))
+      # objective function to optimize
+      learneravg_objfun = function(x, task, measure, avg_weight_fun, data) {
+        # This is the objective function we minimize using nlopt
+        prd = avg_weight_fun(task, unlist(x), data)
+        res = prd$score(measure)
+        if (measure$minimize) res else -res
+      }
 
-  opt = nloptr::nloptr(
-    x0 = rep(1 / n_weights, n_weights),
-    eval_f = nlopt_objfun,
-    lb = rep(0, n_weights), ub = rep(1, n_weights),
-    eval_g_ineq = function(x, task, measure, avg_weight_fun, data) max(x) - 1,
-    opts = list(algorithm = pars$algorithm, xtol_rel = 1e-8),
-    task = task,
-    measure = measure,
-    avg_weight_fun = avg_weight_fun,
-    data = data
-  )$solution
+      pars = self$param_set$get_values(tags = "train")
+      ps = ParamSet$new(params = imap(data, function(x, n) {
+        if (is.numeric(n)) n = paste0("w.", n)
+        ParamDbl$new(id = n, lower = 0, upper = 1)
+      }))
+      optimizer = pars$optimizer
+      if (inherits(optimizer, "character")) {
+        optimizer = bbotk::opt(optimizer)
+        if (inherits(optimizer, "OptimizerNLoptr")) {
+          optimizer$param_set$values = list(xtol_rel = 1e-8, algorithm = "NLOPT_LN_COBYLA", x0 = rep(1/n_weights, n_weights))
+        }
+      }
+      measure = pars$measure
+      if (is.character(measure)) measure = msr(measure)
```

Comment:
if `as_measure()` works here use that

Reply:
There is no `as_measure.character`

Reply:
huh ok, thought this is what `as_measure` was good for.